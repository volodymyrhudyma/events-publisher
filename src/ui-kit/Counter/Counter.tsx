import React from 'react';
import Hint from 'src/ui-kit/Hint';
import { Wrapper } from './styled/Counter.styled';
import { useTranslation } from 'react-i18next';

interface ICounterProps {
  maxLength?: number;
  currentLength?: number;
}

const Counter = ({ maxLength, currentLength }: ICounterProps) => {
  const { t } = useTranslation();
  return (
    <Wrapper>
      <Hint>{t('general.maxLength', { number: maxLength })}</Hint>
      <Hint>
        {currentLength} / {maxLength}
      </Hint>
    </Wrapper>
  );
};

export default Counter;
