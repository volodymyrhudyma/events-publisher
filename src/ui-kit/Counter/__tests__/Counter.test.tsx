import React from 'react';
import { shallow } from 'enzyme';
import Counter from '../Counter';

describe('Counter component', () => {
  it('should render correctly', () => {
    const component = shallow(<Counter maxLength={100} currentLength={0} />);
    expect(component).toMatchSnapshot();
  });
});
