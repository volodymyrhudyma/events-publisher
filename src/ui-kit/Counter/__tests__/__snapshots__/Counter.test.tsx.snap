// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should render correctly 1`] = `
<styled.div>
  <Hint>
    general.maxLength
  </Hint>
  <Hint>
    0
     / 
    100
  </Hint>
</styled.div>
`;
